### PR TITLE
MGMT-2240 adding time calculation to auth, register and update cluster

### PR DESCRIPTION
**Pull request by tsorya**

### CODE REVIEW

**Review by filanov — COMMENTED** [review]
I was just thinking about a log on successful operation at the end and another log at start this way we will be able to see races for example.

**Comment by filanov** [comment]
@tsorya Need to add log in install as well.

**Comment by tsorya** [comment]
@filanov added log to install, not sure regarding successful operation. Let's start and see what we miss

**Comment by tsorya** [comment]
though i can add this line

**Review by filanov — APPROVED** [review]

**Comment by romfreiman** [comment]
please add jira

**Comment by filanov** [comment]
/hold 
@romfreiman Asked to delay it after the release.

File: pkg/auth/auth_handler.go [hunk 1]
```diff
@@ -212,6 +217,10 @@ func (a *AuthHandler) CreateAuthenticator() func(name, in string, authenticate s
 		getToken := func(r *http.Request) string { return r.Header.Get(name) }
 
 		return security.HttpAuthenticator(func(r *http.Request) (bool, interface{}, error) {
+			now := time.Now()
+			defer func() {
+				a.log.Infof("HttpAuthenticator took %v", time.Since(now))
```
Comment (rollandf) [inline pkg/auth/auth_handler.go:222]: Why we have it here and also in `AuthUserAuth` ?

**Review by rollandf — CHANGES_REQUESTED** [review]
Maybe we should look into pprof instead of adding logs...

File: pkg/auth/auth_handler.go (hunk 1 above)
Comment (tsorya) [reply pkg/auth/auth_handler.go:222]: was not sure what was the best func, so added in 2 of them

**Comment by tsorya** [comment]
idea was to make something quick, we will revert it after that. pprof doesn't give us more than just log in this case. We are not monitoring memory and cpu here.

File: pkg/auth/auth_handler.go (hunk 1 above)
Comment (rollandf) [reply pkg/auth/auth_handler.go:222]: So we can have it only here, it will work for both agent and user auth.

**Comment by mhrivnak** [comment]
> idea was to make something quick, we will revert it after that.

It would be helpful for discussion and other purposes to add a description in the jira issue at least of what the goal and expectations are for this. Bonus points for a commit message that describes the scope of the actual change.

**Comment by ronniel1** [comment]
> > idea was to make something quick, we will revert it after that.
> 
> It would be helpful for discussion and other purposes to add a description in the jira issue at least of what the goal and expectations are for this. Bonus points for a commit message that describes the scope of the actual change.

+1

File: go.mod [hunk 2]
```diff
@@ -38,7 +38,6 @@ require (
 	github.com/pborman/uuid v1.2.0
 	github.com/pkg/errors v0.9.1
 	github.com/prometheus/client_golang v1.6.0
-	github.com/prometheus/common v0.9.1
```
[the remark this answers is not in the document]
Comment (tsorya) [reply go.mod:41]: it is not me:) it is golang

File: internal/metrics/metricsManager.go [hunk 3]
```diff
@@ -66,6 +70,7 @@ type MetricsManager struct {
 	serviceLogicClusterCreation              *prometheus.CounterVec
 	serviceLogicClusterInstallationStarted   *prometheus.CounterVec
 	serviceLogicClusterInstallationSeconds   *prometheus.HistogramVec
+	serviceLogicOperationDurationMiliSeconds *prometheus.GaugeVec
```
[the remark this answers is not in the document]
Comment (tsorya) [reply internal/metrics/metricsManager.go:73]: why HIstograp? i think better to see it as Gauge

File: internal/metrics/metricsManager.go [hunk 4]
```diff
@@ -34,6 +35,7 @@ const (
 	counterDescriptionClusterCreation              = "Number of cluster resources created, by version"
 	counterDescriptionClusterInstallationStarted   = "Number of clusters that entered installing state, by version"
 	counterDescriptionClusterInstallationSeconds   = "Histogram/sum/count of installation time for completed clusters, by result and OCP version"
+	counterDescriptionOperationDurationMiliSeconds = "Histogram/sum/count of operation time for specific operation, by name and requestId"
```
Comment (avishayt) [inline internal/metrics/metricsManager.go:38]:
```suggestion
	counterDescriptionOperationDurationMiliSeconds = "Histogram/sum/count of operation time for specific operation, by name"
```

File: pkg/log/context.go [hunk 5]
```diff
@@ -54,3 +57,14 @@ func goid() string {
 	idField := strings.Fields(strings.TrimPrefix(string(buf[:n]), "goroutine "))[0]
 	return idField
 }
+
+func MeasureOperation(operation string, log logrus.FieldLogger, metricsApi metrics.API) func() {
```
Comment (avishayt) [inline pkg/log/context.go:61]: Why put it in log package?  Seems like a very strange place for it.

**Review by avishayt — CHANGES_REQUESTED** [review]
This doesn't include cache hit/miss counters right?

**Comment by tsorya** [comment]
> This doesn't include cache hit/miss counters right?

nope, as we talked will do it in another ticket

File: pkg/log/context.go (hunk 5 above)
Comment (tsorya) [reply pkg/log/context.go:61]: was meant to log this message and then added metrics for it. So not sure where to add it,

Comment (tsorya) [reply pkg/log/context.go:61]: maybe common?

Comment (tsorya) [reply pkg/log/context.go:61]: but we don't have common pkg. Dunno where to add it.

**Review by avishayt — APPROVED** [review]

**Pull request merged**